vm/actor/src/util/set.rs

```diff
@@ -53,10 +58,11 @@ where
 
     /// Deletes key from set.
     #[inline]
-    pub fn delete(&mut self, key: &[u8]) -> Result<(), Error> {
-        self.0.delete(key)?;
-
-        Ok(())
+    pub fn delete(&mut self, key: &[u8]) -> Result<Option<()>, Error> {
```

Comment:
`Option<()>` can definitely be simplified to be a `bool`, can't think of a benefit to having it as an option

Reply:
I like the option because then we can do the method chaining when handling the error instead of checking if its Some(false). What do you think?

Reply:
Then what is the need to erase the return in the function? This may lead to having to do unnecessary work if you need to retrieve the value first?

Reply:
Good catch. gunna return the byteskey

Reply:
Actually... Does that make sense? we should already have the key.... The value of the underlying map is always () because this is a Set that backed by a hamt